Source/ACE/Entity/WorldObject.cs

```diff
@@ -160,101 +160,101 @@ public virtual void SerializeCreateObject(BinaryWriter writer)
             if ((WeenieFlags & WeenieHeaderFlag.PuralName) != 0)
                 writer.WriteString16L(GameData.NamePlural);
 
-            if ((WeenieFlags & WeenieHeaderFlag.ItemCapacity) != 0)
-                writer.Write(GameData.ItemCapacity);
+            if (((WeenieFlags & WeenieHeaderFlag.ItemCapacity) != 0) && GameData.ItemCapacity != null)
```

Comment:
`(WeenieFlags & WeenieHeaderFlag.ItemCapacity) != 0) && GameData.ItemCapacity != null` requires that Weenieflag must be set, and the business object must be a non-null value for the writer to send data.

Ripley is stating that the client may be expecting this, if the flag is sent in a different writer.write packet in the inherited object:

line 149.